api/src/main/java/com/capitalone/dashboard/auth/webhook/github/GithubWebHookRequestFilter.java

```diff
@@ -0,0 +1,132 @@
+package com.capitalone.dashboard.auth.webhook.github;
+
+import com.capitalone.dashboard.auth.AuthenticationResultHandler;
+import com.capitalone.dashboard.settings.ApiSettings;
+import com.capitalone.dashboard.util.HygieiaUtils;
+import com.capitalone.dashboard.webhook.github.GitHubWebHookSettings;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+public class GithubWebHookRequestFilter extends UsernamePasswordAuthenticationFilter {
+    private static final Log LOG = LogFactory.getLog(GithubWebHookRequestFilter.class);
+
+    private final GithubWebHookAuthService githubWebHookAuthService;
+    private final ApiSettings apiSettings;
+
+    protected boolean authenticated = false; // This is to support the unit tests
+
+    public GithubWebHookRequestFilter(String path, AuthenticationManager authManager,
+                                      GithubWebHookAuthService githubWebHookAuthService,
+                                      ApiSettings apiSettings,
+                                      AuthenticationResultHandler authenticationResultHandler) {
+        super();
+        setAuthenticationManager(authManager);
+        setAuthenticationSuccessHandler(authenticationResultHandler);
+        setFilterProcessesUrl(path);
+        this.githubWebHookAuthService = githubWebHookAuthService;
+        this.apiSettings = apiSettings;
+    }
+
+    public void doFilter(ServletRequest req, ServletResponse res, FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest request = (HttpServletRequest)req;
+        HttpServletResponse response = (HttpServletResponse)res;
+
+        String userAgent = request.getHeader("User-Agent");
+        String githubEnterpriseHost = request.getHeader("X-GitHub-Enterprise-Host");
+
+        GitHubWebHookSettings gitHubWebHookSettings = parseAsGitHubWebHook(apiSettings.getGitHubWebHook());
```

Comment:
Unmarshalling of json(apiSettings.getGitHubWebHook()) object is going to happen for every api request which is unnecessary.